src/material-experimental/mdc-chips/chip-grid.ts

```diff
@@ -6,19 +6,434 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/core';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+import {BACKSPACE} from '@angular/cdk/keycodes';
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  DoCheck,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Optional,
+  Output,
+  Self,
+  ViewEncapsulation
+} from '@angular/core';
+import {ControlValueAccessor, FormGroupDirective, NgControl, NgForm} from '@angular/forms';
+import {
+  CanUpdateErrorState,
+  CanUpdateErrorStateCtor,
+  ErrorStateMatcher,
+  mixinErrorState,
+} from '@angular/material/core';
+import {MatFormFieldControl} from '@angular/material/form-field';
+import {MatChipTextControl} from './chip-text-control';
+import {merge, Observable, Subscription} from 'rxjs';
+import {startWith, takeUntil} from 'rxjs/operators';
 
+import {MatChipEvent} from './chip';
+import {MatChipSet} from './chip-set';
+
+
+/** Change event object that is emitted when the chip grid value has changed. */
+export class MatChipGridChange {
+  constructor(
+    /** Chip grid that emitted the event. */
+    public source: MatChipGrid,
+    /** Value of the chip grid when the event was emitted. */
+    public value: any) { }
+}
+
+/**
+ * Boilerplate for applying mixins to MatChipGrid.
+ * @docs-private
+ */
+class MatChipGridBase extends MatChipSet {
+  constructor(_elementRef: ElementRef,
+              _changeDetectorRef: ChangeDetectorRef,
+              public _defaultErrorStateMatcher: ErrorStateMatcher,
+              public _parentForm: NgForm,
+              public _parentFormGroup: FormGroupDirective,
+              /** @docs-private */
+              public ngControl: NgControl) {
+    super(_elementRef, _changeDetectorRef);
+  }
+}
+const _MatChipGridMixinBase: CanUpdateErrorStateCtor & typeof MatChipGridBase =
+    mixinErrorState(MatChipGridBase);
+
+/**
+ * An extension of the MatChipSet component used with MatChipRow chips and
+ * the matChipInputFor directive.
+ */
 @Component({
   moduleId: module.id,
   selector: 'mat-chip-grid',
-  templateUrl: 'chip-grid.html',
+  template: '<ng-content></ng-content>',
   styleUrls: ['chips.css'],
   host: {
-    'class': 'mat-mdc-chip-grid',
+    'class': 'mat-mdc-chip-set mat-mdc-chip-grid mdc-chip-set',
+    'role': 'grid',
+    '[tabIndex]': 'empty ? -1 : 0',
+    // TODO: replace this binding with use of AriaDescriber
+    '[attr.aria-describedby]': '_ariaDescribedby || null',
+    '[attr.aria-required]': 'required.toString()',
+    '[attr.aria-disabled]': 'disabled.toString()',
+    '[attr.aria-invalid]': 'errorState',
+    '[class.mat-chip-list-disabled]': 'disabled',
```

Comment:
Can you add `-mdc-` to these class names? I want to make sure none of the classes overlap with the classes from the non-mdc version. That way we avoid issues resulting from CSS overrides that were aimed at the old implementation breaking the new implementation